src/pages/eventing/EventingLogs.jsx

```diff
@@ -0,0 +1,153 @@
+
+const getIconByStatus = (status) => {
+  switch(status){
+    case "processed":
+      return <Icon type="check" style={{color: "#00FF00"}}/>
+    case "failed":
+      return <Icon type="close" style={{color: "red"}}/>
+    default:
+      return <Icon type="hourglass" />
+  }
+}
+
+function isJson(str) {
+  try {
+      JSON.parse(str);
+  } catch (e) {
+    return str;
+  }
+  const jsonObj = JSON.parse(str)
+  return jsonObj;
+}
+
+const columns = [
+  { title: 'Event ID', dataIndex: '_id', key: '_id' },
+  { title: 'Trigger name', dataIndex: 'rule_name', key: 'rule_name' },
+  { title: 'Invocations', key: 'invocations', render: record => <p>{record.invocation_logs.length}</p> },
+  { title: 'Status', key: 'status', render: record => getIconByStatus(record.status) },
+  { title: 'Created at', key: 'date', render:(record) => <p>{new Date(record.event_timestamp).toISOString()}</p> },
+];
+
+const EventingLogs = () => {
+  // Router params
+  const { projectID } = useParams()
+  const [modalVisible, setModalVisible] = useState(false);
+  const dispatch = useDispatch();
+  const eventLogs = useSelector(state => state.eventLogs);
+  const eventFilters = useSelector(state => state.uiState.eventFilters);
+  const [moreEventLogs, hasMoreEventLogs] = useState(true);
+
+  useEffect(() => {
+    dispatch(increment("pendingRequests"));
+    client.eventing.fetchEventLogs(projectID, eventFilters, 0)
+    .then(res => dispatch(set("eventLogs", res)))
+    .catch(ex => console.log(ex))
+    .finally(() => dispatch(decrement("pendingRequests")))
+  }, [eventFilters])
+  
+  const expandedInvocations = record => {
+    const requestJSON = isJson(record.request_payload);
+    const responseJSON = isJson(record.response_body); 
+    return (
+      <div>
+        <b>Request payload</b><br/>
+        <pre>{JSON.stringify(requestJSON, null, 2)}</pre><br/><br/>
+        <b>Status Code</b><br/>
+        {record.response_status_code}<br/><br/>
+        <b>Response</b><br/>
+        <pre>{JSON.stringify(responseJSON, null, 2)}</pre>
+      </div>
+    )
+  }
+
+  const expandedRowRender = record => {
+    const columns = [
+      { title: 'Status', key: 'status', render: record => {
+        if(record.response_status_code === 200) return <Icon type="check" style={{color: "#00FF00"}}/>
+        else return  <Icon type="close" style={{color: "red"}}/>
+      } },
+      { title: 'ID', dataIndex: '_id', key: '_id' },
+      { title: 'Date', key: "invocation", render: (record) => <p>{record.invocation_time.toString()}</p>}
+    ];
+
+    return (
+    <Table 
+      showHeader={false} 
+      columns={columns} 
+      dataSource={record.invocation_logs} 
+      pagination={false}
+      expandedRowRender={expandedInvocations} 
+      title={() => 'Invocations'} 
+    /> );
+  }
+
+  const filterTable = (values) => {
+     console.log(values)
+     dispatch(set("uiState.eventFilters", values))
+     hasMoreEventLogs(true);
```

Comment:
Do you think this logic is correct? After applying a filter, the event logs result should be overwritten rather than concated because the previous event logs in the redux array might contain results which does not fulfill filters value.